src/shared/sam/localLambdaRunner.ts

```diff
@@ -65,7 +65,11 @@ function getEnvironmentVariables(
  * Decides the resource name for the generated template.yaml.
  */
 function makeResourceName(config: SamLaunchRequestArgs): string {
-    return config.invokeTarget.target === 'code' ? 'awsToolkitSamLocalResource' : config.invokeTarget.logicalId
+    return config.invokeTarget.target === 'code'
+        ? config.runtimeFamily === RuntimeFamily.Java
```

Comment:
Once the `?:` cascade can we split it into an if else?

Reply:
agreed in this case. it might help explain what's going on. Is the reason for this, because we don't make a temporary project for java? That could be explained here.

Reply:
> Alternately, we can shift everything to using the project root name. I'm not sure if there's a reason why we didn't do that before.

I doubt there was a reason. +1 for doing that.